feat: 핸드폰 인증하기 플로우 완료

ChaNoo97 opened the pull request:
- email 인증 화면 (비어있는 화면) 으로 넘어가게끔 해놓음

## Motivation ⍰

- 핸드폰 인증하기 화면

<br>

## Key Changes 🔑

- 인증하기 화면 부분 input output 패턴 재정비
- 인증완료 이후 lottie popup 완료
- popup 이후 callback 이용, input trigger 에 vc push 하는 로직 추가

<br>

## To Reviewers 🙏🏻

- [ ] lottie 에서 callback 받아오는 부분에 대해 의문이 있습니다. 이런 패턴이 맞을지 한번 확인부탁 드립니다.
- [ ] input output 패턴 이용시 viewModel 의 transform 에서는 dispose 되지 않도록 하는것 이라고 알고 있는데, navigator 을 이용할 시 dispose 해줘야하는 input stream 이 있는것으로 보입니다 이부분에 대한 논의가 필요해 보입니다.

<br>

## Linked Issue 🔗

- https://github.com/THT-Team/THT-iOS/issues/12#issue-1838190900

ChaNoo97 committed:
feat: 핸드폰 인증하기 플로우 완료

- email 인증 화면 (비어있는 화면) 으로 넘어가게끔 해놓음

ChaNoo97 committed:
fix: color 이름 변경된 것 추가 수정

ChaNoo97 committed:
chor: 파일, dict 명 수정

ibcylon committed:
Refactor remove class member and add tranform method local member

viewType, timerRelay, authNumber 제거
로직 뷰모델에서 struct로 옮김 test를 위해서..?

ibcylon committed:
Refactor: remove viewType

ibcylon commented:
기존 뷰모델 클래스 멤버였던 viewType(삭제), timerRelay(삭제), authNumber(이동)을 transform 로컬 변수로 옮겨서 VM.transform() 함수 내용 좀 변경했습니다. 

작성하신 AuthCodeWithTimeStamp Struct를 순수하게 데이터만 담고 있는 걸로 놔둘지 아니면 로직을 포함하는게 좋을 지 한 번 봐주세요!

Minny27 committed:
Update: 핸드폰 인증 UI Test 코드 추가

- Project 파일에 타겟 추가
- VC에 setupAccessibilityIdentifier() 함수 추가 및 호출
- LoginTest 파일 추가

Minny27 commented:
UI Test 타겟 추가해서 핸드폰 인증에서 폰 번호 입력하는 것까지 코드 작성해놨습니다.
추가적으로 토큰으로 로그인 체크를 키체인에서 체크하는 방식으로 해놨고 메인 쪽 테스트할 때는 delete 코드 주석처리하시면 될 거 같습니다.
<Img src="https://github.com/THT-Team/THT-iOS/assets/68800789/1680a67a-534d-4a21-a2fc-6068374f6fb8" width=60%>
